.github/workflows/library-release.yml

```diff
@@ -1,23 +1,116 @@
-name: Build and Release
+name: Publish c2pa-library binaries
 
 on:
   push:
-    branches: [ "gpeacock/c2pa_c" ]
     tags:
       - 'c2pa-v*' # Trigger on version tags (e.g., v1.0.0)
   workflow_dispatch: # Allow manual triggering of the workflow
 
-env:
-  CARGO_TERM_COLOR: always
-
 jobs:
   build:
+    name: Publish c2pa-library binaries
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-latest, ubuntu-latest, windows-latest ]
+        rust_version: [ stable ]
+        experimental: [ false ]
+        include:
+          - os: macos-latest
+            target: aarch64-apple-darwin
 
-    runs-on: ubuntu-latest
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
 
+      - name: Setup Strawberry Perl
+        if: matrix.os == 'windows-latest'
```

Comment:
Opinion: The fact that windows needs Perl, one of the only write-only languages, is quite a thing.

Reply:
I share your opinion of Perl ;)